experiment/httphost/httphost.go

```diff
@@ -0,0 +1,94 @@
+// Package httphost contains the httphost network experiment.
+//
+
+package httphost
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/ooni/probe-engine/experiment/urlgetter"
+	"github.com/ooni/probe-engine/model"
+)
+
+const (
+	testName    = "http_host_header"
+	testVersion = "0.1.0"
+)
+
+// Config contains the experiment config.
+type Config struct {
+	// TestHelperURL is the address of the test helper.
+	TestHelperURL string
+}
+
+// Subresult contains the keys of a single measurement
+// that targets either the target or the control.
+type Subresult struct {
+	urlgetter.TestKeys
+	THAddress string `json:"th_address"`
+}
+
+// TestKeys contains httphost test keys.
+type TestKeys struct {
+	Control Subresult `json:"control"`
+	Result  string    `json:"result"`
+	Target  Subresult `json:"target"`
+}
+
+// Measurer performs the measurement.
+type Measurer struct {
+	config Config
+}
+
+// ExperimentName implements ExperimentMeasurer.ExperiExperimentName.
+func (m *Measurer) ExperimentName() string {
+	return testName
+}
+
+// ExperimentVersion implements ExperimentMeasurer.ExperimentVersion.
+func (m *Measurer) ExperimentVersion() string {
+	return testVersion
+}
+
+// Run implements ExperimentMeasurer.Run.
+func (m *Measurer) Run(
+	ctx context.Context,
+	sess model.ExperimentSession,
+	measurement *model.Measurement,
+	callbacks model.ExperimentCallbacks,
+) error {
+	if measurement.Input == "" {
+		return errors.New("Experiment requires measurement.Input")
+	}
+	if m.config.TestHelperURL == "" {
+		return errors.New("Experiment requires test helper address.")
+	}
+	urlgetter.RegisterExtensions(measurement)
+	// perform the measurement
+	g := urlgetter.Getter{
+		Begin: measurement.MeasurementStartTimeSaved,
```

Comment:
I like this idea of getting `Begin` from the measurement directly, thanks!

Reply:
I think I might have just copied it from elsewhere  ... but thanks :)